tool.go

```diff
@@ -121,13 +121,22 @@ func main() {
 
 				for _, pkgPath := range args {
 					pkgPath = filepath.ToSlash(pkgPath)
-					if s.Watcher != nil {
-						s.Watcher.Add(pkgPath)
-					}
 					pkg, err := gbuild.Import(pkgPath, 0, s.InstallSuffix(), options.BuildTags)
 					if err != nil {
 						return err
 					}
+					if s.Watcher != nil {
+						s.Watcher.Add(pkg.Dir)
+
+						// We want to have the package dir watched when calling Import to avoid
+						// races where we could miss a change made and not re-trigger a build.
```

Comment:
Yes, the reason for calling import a second time is because a file may change after Import has processed it but before the file is watched. The reason the first Import is fine is that the build isn't being done based on it (since it's replaced by the second Import). If a file changes during the first import then the second import will use the changed file, and if a file changes during the second Import, then the Watcher will trigger another build.